v2/pkg/genruntime/kubernetes_config_exporter.go

```diff
@@ -14,12 +14,12 @@ import (
 	"github.com/Azure/azure-service-operator/v2/internal/genericarmclient"
 )
 
-// KubernetesExporter defines a resource which can create other resources in Kubernetes.
-type KubernetesExporter interface {
-	// ExportKubernetesResources provides a list of Kubernetes resource for the operator to create once the resource which
+// KubernetesConfigExporter defines a resource which can create configmaps in Kubernetes
+type KubernetesConfigExporter interface {
+	// ExportKubernetesConfigMaps provides a list of Kubernetes ConfigMaps for the operator to create once the resource which
 	// implements this interface is successfully provisioned. This method is invoked once a resource has been
 	// successfully created in Azure, but before the Ready condition has been marked successful.
-	ExportKubernetesResources(
+	ExportKubernetesConfigMaps(
```

Comment:
This function actually can still export any Kubernetes resource, we just have a special function for secrets.

I could keep this as `KubernetesExporter` rather than making it ConfigMap specific. Not sure if that's better?